File: ParkPal.Application/src/models/api/Destination.ts
import Park from "@/models/api/Park";
import axios from "axios";

export default class Destination {
    destinationId = '';
    name = '';
    image?: string;
    location?: string;
    parks: Array<Park> = new Array<Park>();
    hidden?: boolean;
    defaultOrder?: number;
    visible = !this.hidden;


    constructor(data: Pick<Destination, "destinationId" | "name" | "image" | "location" | "parks" | "hidden" | "defaultOrder"> | null = null) {
        if(data != null) {
            this.destinationId = data.destinationId;
            this.name = data.name;
            this.image = data.image;
            this.location = data.location;
            this.hidden = data.hidden;
            this.defaultOrder = data.defaultOrder;
            this.visible = !data.hidden;

            data.parks?.forEach((park: Park) => {
                this.parks.push(new Park(park))
            });
        }
    }

    checkImageExists() {
        return new Promise((resolve) => {
            if(this.image) {
                axios.get('/img/' + this.image).then(() => {
                    resolve(true);
                }).catch(() => {
                    resolve(false);
                });
            }else{
                resolve(true);
            }
        })
    }
}

File: ParkPal.Common/Models/Destination.cs
namespace ParkPal.Common.Models;

public class Destination
{
    public string DestinationId { get; set; }
    public string Name { get; set; }
    public string Timezone { get; set; }
    public double? Longitude { get; set; }
    public double? Latitude { get; set; }
    

    public List<Park> Parks { get; set; }

    public Destination(string destinationId, string name)
    {
        DestinationId = destinationId;
        Name = name;
        Parks = new List<Park>();
    }
}